featherweb/__init__.py

```diff
@@ -35,56 +36,66 @@ def run(self, timeout=5, callback=None, **kwargs):
         poller = select.poll()
         poller.register(self.m_Socket, select.POLLIN)
 
-        while True:
-            events = poller.poll(timeout*1000)
-            if not events and callback:
-                if not callback(**kwargs):
-                    break
-                continue
+        running = True
+        while running:
+            try:
+                events = poller.poll(timeout*1000)
+                if not events and callback:
+                    if not callback(**kwargs):
```

Comment:
Maybe we should set running to the return value of the callback?  Maybe that would be easier to read?

Reply:
And the colon removed too...

Reply:
Oops²! :)